### DISCUSSION TITLE
libpulse0 fails to fetch with apt (repo broken?)

### DISCUSSION THREAD

**Discussion by vadi2** [community/community#26455] [Actions]
<p>Quite a few of my builds are failing to fetch libpulse0 which is in <a href="https://packages.ubuntu.com/search?keywords=libpulse0&amp;searchon=names&amp;suite=bionic&amp;section=all" rel="nofollow noopener">bionic repos</a>:</p>
<pre><code>Err:37 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libpulse0 amd64 1:11.1-1ubuntu7.4
  404 Not Found [IP: 52.177.174.250 80]
Get:38 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libpulse-mainloop-glib0 amd64 1:11.1-1ubuntu7.4 [22.1 kB]
Get:39 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libpulse-dev amd64 1:11.1-1ubuntu7.4 [81.5 kB]
Get:40 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxcb-xkb1 amd64 1.13-2~ubuntu18.04 [30.1 kB]
Get:41 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxkbcommon-x11-0 amd64 0.8.2-1~ubuntu18.04.1 [13.4 kB]
Get:42 http://ppa.launchpad.net/ondrej/php/ubuntu bionic/main amd64 libzip-dev amd64 1.5.1-4+ubuntu18.04.1+deb.sury.org+2 [158 kB]
Fetched 2364 kB in 0s (5153 kB/s)
E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/p/pulseaudio/libpulse0_11.1-1ubuntu7.4_amd64.deb 404 Not Found [IP: 52.177.174.250 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
##[error]Process completed with exit code 100.
</code></pre>
<p>Is the apt repository corrupted?</p>

**Comment by vadi2** [comment]
<p>This isn’t a consistent error either, as re-running the action fixes this. But whenever it fails, it’s always with the <code>libpulse0</code> package.</p>

**Comment by vadi2** [comment]
<p>This happens frequently, here is <a href="https://github.com/Mudlet/Mudlet/pull/3668/checks?check_run_id=611004233" rel="nofollow noopener">another run</a> that failed:</p>
<pre><code>Get:36 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libsndfile1 amd64 1.0.28-4ubuntu0.18.04.1 [170 kB]
Err:37 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libpulse0 amd64 1:11.1-1ubuntu7.4
  404 Not Found [IP: 52.177.174.250 80]
Err:38 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libpulse-mainloop-glib0 amd64 1:11.1-1ubuntu7.4
  404 Not Found [IP: 52.177.174.250 80]
Err:39 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libpulse-dev amd64 1:11.1-1ubuntu7.4
  404 Not Found [IP: 52.177.174.250 80]
Get:40 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxcb-xkb1 amd64 1.13-2~ubuntu18.04 [30.1 kB]
Get:41 http://azure.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxkbcommon-x11-0 amd64 0.8.2-1~ubuntu18.04.1 [13.4 kB]
Get:42 http://ppa.launchpad.net/ondrej/php/ubuntu bionic/main amd64 libzip-dev amd64 1.5.1-4+ubuntu18.04.1+deb.sury.org+2 [158 kB]
E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/p/pulseaudio/libpulse0_11.1-1ubuntu7.4_amd64.deb 404 Not Found [IP: 52.177.174.250 80]
Fetched 2261 kB in 0s (4701 kB/s)
E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/p/pulseaudio/libpulse-mainloop-glib0_11.1-1ubuntu7.4_amd64.deb 404 Not Found [IP: 52.177.174.250 80]
E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/p/pulseaudio/libpulse-dev_11.1-1ubuntu7.4_amd64.deb 404 Not Found [IP: 52.177.174.250 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/vadi2">@vadi2</a> ,</p>
<p>Please try to add the " <strong>sudo apt update</strong>" command before the " <strong>sudo apt install</strong>" command to see if the problem can be solved.</p>
<pre><code>- name: (Linux) Install apt dependencies
      if: runner.os == 'Linux'
      run: |
        sudo apt update
        sudo apt install pkg-config libzip-dev libglu1-mesa-dev libpulse-dev libxkbcommon-x11-0
</code></pre>
<p>The APT package index is basically a database that holds records of available packages from the repositories enabled in your system. I recommend you always update the package index before upgrading or installing new packages.</p>
<p>apt actually works on a database of available packages. If the database is not updated, the system won’t know if there are any newer packages available. This is why updating the repository should be the first thing to do in any Linux system after a fresh install.</p>

**Comment by vadi2** [comment]
<p>That’s what we ended up doing and it works.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/vadi2">@vadi2</a> ,</p>
<p>Glad that the problem has been solved.</p>
<p>If you have any other question about this ticket, feel free to tell us.</p>
